pkg/agent/zeroclient.go

```diff
@@ -0,0 +1,85 @@
+package agent
+
+import (
+	"context"
+	"net"
+	"net/url"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/client"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/client/versions"
+	"github.com/openshift/assisted-service/models"
+
+	"github.com/openshift/installer/pkg/agent/manifests"
+)
+
+type nodeZeroClient struct {
+	restClient          *client.AssistedInstall
+	NodeZeroIP          string
+	clusterZeroID       *strfmt.UUID
+	clusterZeroInfraEnv *strfmt.UUID
+}
+
+type clusterZeroMetadata struct {
+	zeroInfraEnv  string
+	zeroClusterID string
+}
+
+func NewNodeZeroClient() (*nodeZeroClient, error) {
+	ctx := context.Background()
+	n := manifests.NewNMConfig()
+	nodeZeroIP := n.GetNodeZeroIP()
+
+	restConfig := client.Config{}
+	*restConfig.URL = url.URL{
+		Scheme: "http",
+		Host:   net.JoinHostPort(nodeZeroIP, "8090"),
+		Path:   client.DefaultBasePath,
+	}
+	restClient := client.New(restConfig)
+
+	// GET /v2/clusters
+	listClusterParams := installer.NewV2ListClustersParams()
+	clusterResult, err := restClient.Installer.V2ListClusters(ctx, listClusterParams)
+	if err != nil {
+		return nil, err
+	}
+	clusterZeroID := clusterResult.Payload[0].ID
+
+	// GET /v2/infraenvs
+	listInfraEnvParams := &installer.ListInfraEnvsParams{ClusterID: clusterZeroID}
+	infraenvResult, err := restClient.Installer.ListInfraEnvs(ctx, listInfraEnvParams)
+	if err != nil {
+		return nil, err
+	}
+	clusterZeroInfraEnv := infraenvResult.Payload[0].ID
```

Comment:
Keeping the two above GET calls within the factory method could be harder for the maintenance and tests, if possible I'd suggest moving them in a type method to be called separately after client instantiation

Reply:
Yeah I agree with this. I think I can refactor this some. 🤔 

I would like to keep the metadata of ClusterZeroID and ClusterZeroInfraenv in nodeZeroClient to avoid having to call these calls more than once. The goal of storing that metadata was because other API calls ask for ClusterID and Infraenv a lot.

Reply:
Also perhaps nodeZeroClient and clusterZero should be two different structs. Since I'm trying to solve two different problems there.